server/src/main/java/com/datastrato/gravitino/server/web/rest/FilesetOperations.java

```diff
@@ -0,0 +1,186 @@
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.FilesetResponse;
+import com.datastrato.gravitino.dto.util.DTOConverters;
+import com.datastrato.gravitino.file.Fileset;
+import com.datastrato.gravitino.file.FilesetChange;
+import com.datastrato.gravitino.metrics.MetricNames;
+import com.datastrato.gravitino.server.web.Utils;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("metalakes/{metalake}/catalogs/{catalog}/schemas/{schema}/filesets")
+public class FilesetOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(FilesetOperations.class);
+
+  private final CatalogOperationDispatcher dispatcher;
+
+  @Context private HttpServletRequest httpRequest;
+
+  @Inject
+  public FilesetOperations(CatalogOperationDispatcher dispatcher) {
+    this.dispatcher = dispatcher;
+  }
+
+  @GET
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "list-fileset." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "list-fileset", absolute = true)
+  public Response listFilesets(
+      @PathParam("metalake") String metalake,
+      @PathParam("catalog") String catalog,
+      @PathParam("schema") String schema) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            Namespace filesetNS = Namespace.ofFileset(metalake, catalog, schema);
+            NameIdentifier[] idents = dispatcher.listFilesets(filesetNS);
+            return Utils.ok(new EntityListResponse(idents));
+          });
+
+    } catch (Exception e) {
+      return ExceptionHandlers.handleFilesetException(OperationType.LIST, "", schema, e);
+    }
+  }
+
+  @POST
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "create-fileset." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "create-fileset", absolute = true)
+  public Response createFileset(
+      @PathParam("metalake") String metalake,
+      @PathParam("catalog") String catalog,
+      @PathParam("schema") String schema,
+      FilesetCreateRequest request) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            request.validate();
+            NameIdentifier ident =
+                NameIdentifier.ofFileset(metalake, catalog, schema, request.getName());
+
+            Fileset fileset =
+                dispatcher.createFileset(
+                    ident,
+                    request.getComment(),
+                    request.getType(),
```

Comment:
Here `request.getType()` could be null, and it will lead an error when calling `dispatcher.createFileset`.